lib/api/core/models/repositories/pluginRepository.js

```diff
@@ -0,0 +1,94 @@
+const
+  _ = require('lodash'),
+  Repository = require('./repository'),
+  Data = require('../plugin/data');
+
+class PluginRepository extends Repository {
+  constructor (kuzzle, pluginName) {
+    super(kuzzle, pluginName);
+
+    // We set the collection to null on purpose to force plugins
+    // to specify the collection in each calls
+    this.collection = null;
+    this.ObjectConstructor = Data;
+  }
+
+  init (options) {
+    if (options instanceof Object && !Array.isArray(options) && options.cacheEngine !== null) {
+      options.cacheEngine = null;
+    }
+
+    super.init(options);
+  }
+
+  /**
+   * Serializes the object before being persisted to database.
+   *
+   * @param {object} data - The object to serialize
+   * @returns {object}
+   */
+  serializeToDatabase (data) {
+    // avoid the data var mutation
+    const result = _.assignIn({}, data);
```

Comment:
nested properties are shallow copied when using `assignxx`, which contradicts the comment about mutation.
In this case, it does not matter much as the services consuming the result object are not supposed to mutate its internal properties.